tests/testsuite/clippy.rs

```diff
@@ -0,0 +1,79 @@
+use crate::support::{is_nightly, process, project};
+
+#[cargo_test]
+fn clippy() {
```

Comment:
I would prefer to not move this test, and keep the cache-messages tests all in the same file.  And the `fix_with_clippy` can just live in `fix.rs`.  I realize that module taxonomy hierarchies are not perfect, but at least for me it fits better.  Also, the substring "clippy" is unique enough that you can still filter on it.

Reply:
I just got confused when you said "If you do end up adding a test, it would be good to move that "is clippy available" out somewhere." I didn't realize you meant for me to move the logic that checks for clippy-driver being available to a common function rather than move the entire test. mb

Reply:
Ah, sorry for the confusion!